pkg/app/component.go

```diff
@@ -274,6 +283,11 @@ func (c *Compo) mount(d Dispatcher) error {
 			Tag("kind", c.Kind())
 	}
 
+	if Initialiser, ok := c.self().(Initialiser); ok && c.disp == nil {
+		// will only run once even with multiple dismount/mounts
```

Comment:
No need for the comment since if a component is mounted, an error is returned.

Reply:
That is why there is the comment. The component is getting mounted again after being dismounted before.